### PR TITLE
Add chart plugin

### PR DESCRIPTION
**Pull request by Zerquix18**
Better late than never! It's been a busy month for me.

This continues #596  
Addresses the feedback [here](https://github.com/snapshot-labs/snapshot/pull/596#issuecomment-932775711).

### CODE REVIEW

File: src/components/Plugin/Charts/CustomBlock.vue [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      loadVotingPowerPerDayChart(Chart);
+    }
+  },
+  { flush: 'post' }
+);
+</script>
+
+<template>
+  <Block :title="$t('charts.charts')">
+    <div class="py-3 flex flex-row">
```
Comment (mktcode) [inline src/components/Plugin/Charts/CustomBlock.vue:193]:
```suggestion
    <div class="flex flex-row justify-between -m-4 border-b">
```

File: src/components/Plugin/Charts/CustomBlock.vue [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        v-if="space.plugins?.charts.total_votes_per_day.enabled"
+        class="block px-4 py-2 sidenav-item"
+        :class="currentTab === 'total_votes_per_day' && 'charts-selected-tab'"
+        @click="currentTab = 'total_votes_per_day'"
+      >
+      {{ $t('charts.totalVotesPerDay') }}
+      </button>
+      <button
+        v-if="space.plugins?.charts.voting_power_per_address.enabled"
+        class="block px-4 py-2 sidenav-item"
```
Comment (mktcode) [inline src/components/Plugin/Charts/CustomBlock.vue:204]:
```suggestion
        class="block px-4 py-2 sidenav-item w-[150px]"
```

File: src/components/Plugin/Charts/CustomBlock.vue [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        :class="currentTab === 'share_of_voting_power' && 'charts-selected-tab'"
+        @click="currentTab = 'share_of_voting_power'"
+      >
+        {{ $t('charts.shareOfVotingPower') }}
+      </button>
+      <button
+        v-if="space.plugins?.charts.voting_power_per_day.enabled"
+        class="block px-4 py-2 sidenav-item"
```
Comment (mktcode) [inline src/components/Plugin/Charts/CustomBlock.vue:212]:
```suggestion
        class="block px-4 py-2 sidenav-item w-[150px]"
```

File: src/components/Plugin/Charts/CustomBlock.vue [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        :class="currentTab === 'voting_power_per_day' && 'charts-selected-tab'"
+        @click="currentTab = 'voting_power_per_day'"
+      >
+        {{ $t('charts.votingPowerPerDay') }}
+      </button>
+    </div>
+    <div v-if="votes.length > 0">
```
Comment (mktcode) [inline src/components/Plugin/Charts/CustomBlock.vue:219]:
```suggestion
    <div v-if="votes.length > 0" class="mt-6">
```

File: src/components/Plugin/Charts/CustomBlock.vue [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        <canvas ref="totalVotesPerDayChart" />
+      </div>
+      <div v-if="space.plugins?.charts.voting_power_per_address.enabled && currentTab === 'share_of_voting_power'">
+        <canvas ref="votingPowerPerAddress" />
+      </div>
+      <div v-if="space.plugins?.charts.voting_power_per_day.enabled && currentTab === 'voting_power_per_day'">
+        <canvas ref="votingPowerPerDay" />
+      </div>
+    </div>
+    <div v-if="votes.length === 0">
```
Comment (mktcode) [inline src/components/Plugin/Charts/CustomBlock.vue:230]:
```suggestion
    <div v-if="votes.length === 0" class="mt-6">
```

**Review by mktcode — CHANGES_REQUESTED** [review]

**Comment by mktcode** [comment]
@Zerquix18 @bonustrack  With my suggested changes the tabs look like this:

image
image

**Comment by bonustrack** [comment]
@mktcode I like that!

**Review by mktcode — APPROVED** [review]

**Review by bonustrack — APPROVED** [review]

**Pull request merged**